### PR TITLE
NuGet: Set EnableWindowsTargeting as true

### LINKED ISSUE #8572: Dependabot fails to find dependencies in some windows projects (NETSDK1100 / EnableWindowsTargeting error)
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

nuget

### Package manager version

_No response_

### Language version

_No response_

### Manifest location and content before the Dependabot update

Broken file.csproj extract:

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
    <Platforms>AnyCPU;x64</Platforms>
    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="DynamicData" Version="8.1.1" />
  </ItemGroup>
</Project>
```

Working csproj extract:
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net6.0-windows</TargetFramework>
    <IsPackable>false</IsPackable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="DynamicData" Version="8.1.1" />
  </ItemGroup>
</Project>
```

### dependabot.yml content

_No response_

### Updated dependency

_No response_

### What you expected to see, versus what you actually saw

I expected to see dependabot update DynamicData in both `.scproj` files, however only one was updated. 

The broken projects appear to be those that target `net6.0-windows10.0.19041.0`. The log indicates a build is failing for these with error `NETSDK1100: To build a project targeting Windows on this operating system, set the EnableWindowsTargeting property to true`. Then dependabot incorrectly states `Package [DynamicData] Does not exist as a dependency`.

We have tried setting `EnableWindowsTargeting` via `Directory.Build.props` files and this made no difference. We have confirmed that we can build our solution on linux with this property set. 

### Native package manager behavior

_No response_

### Images of the diff or a link to the PR, issue, or logs

Log extract for broken `.csproj` update:
```
updater | running NuGet updater:
updater | /opt/nuget/NuGetUpdater/NuGetUpdater.Cli update --repo-root /home/dependabot/dependabot-updater/repo --solution-or-project /home/dependabot/dependabot-updater/repo/path/to/proj/file.csproj --dependency DynamicData --new-version 8.3.25 --previous-version 8.1.1  --verbose
updater |   No global.json files found.
updater |   No dotnet-tools.json files found.
updater | Running for project [/home/dependabot/dependabot-updater/repo/path/to/proj/file.csproj]
updater |   Running for SDK-style project
updater | dotnet build in GetAllPackageDependenciesAsync failed. STDOUT: MSBuild version 17.6.3+07e294721 for .NET
updater |   Determining projects to restore...
updater | /usr/local/dotnet/current/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1100: To build a project targeting Windows on this operating system, set the EnableWindowsTargeting property to true. [/tmp/package-dependency-resolution_SAEhSC/Project.csproj]
updater |
updater | Build FAILED.
updater |
updater | /usr/local/dotnet/current/sdk/7.0.305/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(90,5): error NETSDK1100: To build a project targeting Windows on this operating system, set the EnableWindowsTargeting property to true. [/tmp/package-dependency-resolution_SAEhSC/Project.csproj]
updater |     0 Warning(s)
updater |     1 Error(s)
updater |
updater | Time Elapsed 00:00:03.53
updater |
updater |  STDERR:
updater |
updater |     Package [DynamicData] Does not exist as a dependency in [/home/dependabot/dependabot-updater/repo/path/to/proj/file.csproj].
updater | Update complete.
```

### Smallest manifest that reproduces the issue

this is a full manifest from a minimal example project that shows this issue:
```
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <OutputType>WinExe</OutputType>
        <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
        <Nullable>enable</Nullable>
        <UseWPF>true</UseWPF>
    </PropertyGroup>
    
    <ItemGroup>
        <PackageReference Include="DynamicData" Version="8.1.1" />
    </ItemGroup>
</Project>
```

### PR DESCRIPTION
**Pull request by na1307**
Fixes: #8971, Fixes #8572

This prevents [NETSDK1100](https://learn.microsoft.com/en-us/dotnet/core/tools/sdk-errors/netsdk1100) errors from occurring for Windows-specific apps.

### CODE REVIEW

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs [hunk 1]
```diff
@@ -375,7 +375,7 @@ private static ProjectRootElement CreateProjectRootElement(ProjectBuildFile buil
         {
             var tempProjectPath = await CreateTempProjectAsync(tempDirectory, repoRoot, projectPath, targetFramework, packages);
 
-            var (exitCode, stdout, stderr) = await ProcessEx.RunAsync("dotnet", $"build \"{tempProjectPath}\" /t:_ReportDependencies");
+            var (exitCode, stdout, stderr) = await ProcessEx.RunAsync("dotnet", $"build \"{tempProjectPath}\" /t:_ReportDependencies /p:EnableWindowsTargeting=true");
```
Comment (JoeRobich) [inline nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs:378]: Could we move this property into the Directory.Build.props that we write out in the `CreateTempProjectAsync` method? That method is used elsewhere and I want to be sure everyone gets the benefit.

**Review by JoeRobich — COMMENTED** [review]
I love that it is such a simple fix.

File: nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs (hunk 1 above)
Comment (na1307) [reply nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs:378]: Okay, what should I do? Should I change line 363?

Comment (JoeRobich) [reply nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Utilities/MSBuildHelper.cs:378]: That's right. You could do something like the following.

```c#
        await File.WriteAllTextAsync(
            Path.Combine(tempDir.FullName,
            "Directory.Build.props"),
            """
            <Project>
              <PropertyGroup>
                <EnableWindowsTargeting>true</EnableWindowsTargeting>
              </PropertyGroup>
            </Project>
            """);
```

**Comment by JoeRobich** [comment]
@na1307 Could you add a test case for this to UpdateWorkerTests.SDK.cs?

Something like this should work, based on code sample in https://github.com/dependabot/dependabot-core/issues/8572
```c#
        [Fact]
        public async Task UpdateVersionAttribute_InProjectFile_ForPackageReferenceInclude_Windows()
        {
            // update Newtonsoft.Json from 9.0.1 to 13.0.1
            await TestUpdateForProject("Newtonsoft.Json", "9.0.1", "13.0.1",
                // initial
                projectContents: $"""
                <Project Sdk="Microsoft.NET.Sdk">
                  <PropertyGroup>
                    <TargetFramework>net8.0-windows10.0.19041.0</TargetFramework>
                    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
                  </PropertyGroup>
                
                  <ItemGroup>
                    <PackageReference Include="Newtonsoft.Json" Version="9.0.1" />
                  </ItemGroup>
                </Project>
                """,
                // expected
                expectedProjectContents: $"""
                <Project Sdk="Microsoft.NET.Sdk">
                  <PropertyGroup>
                    <TargetFramework>net8.0-windows10.0.19041.0</TargetFramework>
                    <RuntimeIdentifier>win10-x64</RuntimeIdentifier>
                  </PropertyGroup>
                
                  <ItemGroup>
                    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
                  </ItemGroup>
                </Project>
                """);
        }
```

**Review by JoeRobich — APPROVED** [review]
Thanks @na1307!

**Comment by Nishnha** [comment]
Hi I'm deploying this

**Pull request merged**